config/config.go

```diff
@@ -166,14 +223,327 @@ func (cs configs) ResolveConfig(f func(configs []interface{}) error) error {
 // The default configuration sources are:
 // * Environment Variables
 // * Shared Configuration and Shared Credentials files.
-func LoadDefaultConfig(cfgs ...Config) (aws.Config, error) {
-	var cfgCpy configs
-	cfgCpy = append(cfgCpy, cfgs...)
+func LoadDefaultConfig(ctx context.Context, optFns ...func(*LoadOptions) error) (aws.Config, error) {
+	var cfg aws.Config
+	options := LoadOptions{}
+
+	for _, optFn := range optFns {
+		optFn(&options)
+	}
 
-	cfgCpy, err := cfgCpy.AppendFromLoaders(defaultLoaders)
+	// assign Load Options to configs
+	var cfgCpy = configs{options}
+
+	cfgCpy, err := cfgCpy.AppendFromLoaders(ctx, defaultLoaders)
+	if err != nil {
+		return aws.Config{}, err
+	}
+
+	cfg, err = cfgCpy.ResolveAWSConfig(ctx, defaultAWSConfigResolvers)
 	if err != nil {
 		return aws.Config{}, err
 	}
 
-	return cfgCpy.ResolveAWSConfig(defaultAWSConfigResolvers)
+	return cfg, nil
+}
+
+// Getter functions to satisfy interfaces
+
+// getRegion returns Region from config's LoadOptions
+func (o LoadOptions) getRegion(context.Context) (string, error) {
+	if len(o.Region) != 0 {
+		return o.Region, nil
+	}
+
+	return o.DefaultRegion, nil
+}
+
+// WithRegion is a helper function to construct functional options
+// that sets Region on config's LoadOptions.
+func WithRegion(v string) func(*LoadOptions) error {
+	return func(o *LoadOptions) error {
+		o.Region = v
+		return nil
+	}
+}
+
+// getDefaultRegion returns DefaultRegion from config's LoadOptions
+func (o LoadOptions) getDefaultRegion(ctx context.Context) (string, bool, error) {
+	if len(o.DefaultRegion) == 0 {
+		return o.DefaultRegion, false, nil
+	}
+
+	return o.DefaultRegion, true, nil
+}
```

Comment:
minor styling nit, the implementation for default region is inverse of getRegion.  == 0 vs != 0.